fancy/analysis/analysis.py

```diff
@@ -965,22 +966,23 @@ def _prepare_fit_inputs(self):
             ptype = str(self.model.ptype)
             _, self.fit_input["Z"] = self.nuc_table[ptype]
 
-        if self.analysis_type == self.gmf_type or self.analysis_type == self.composition_type:
+        if self.analysis_type == self.gmf_type or self.analysis_type == self.composition_type or self.analysis_type == self.GMFdirection_type:
             self.fit_input["kappa_gmf"] = self.data.uhecr.kappa_gmf
             # self.fit_input["kappa_gmf"] = np.ones_like(
-            #     self.data.uhecr.kappa_gmf) * self.data.detector.kappa_d
+            #     self.data.uhecr.kappa_gmf) * self.data.detector.kappa
         else:
             self.fit_input["kappa_d"] = self.data.detector.kappa_d
 
-        if self.analysis_type == self.composition_type:
+        if self.analysis_type == self.composition_type or self.analysis_type == self.GMFdirection_type:
             # interpolation grid for theta & kappa
             self.fit_input["NthetaP"] = len(self.gmf_deflections._thetaP_grid)
             self.fit_input["thetaP_grid"] = np.rad2deg(self.gmf_deflections._thetaP_grid)
             self.fit_input["kappa_interp_grid"] = self.gmf_deflections._f_kappa(self.gmf_deflections._thetaP_grid)
 
             # add rigidity instead of energy
-            self.fit_input["Rdet"] = self.data.uhecr.energy / self.Zdet
-            self.fit_input["Rth"] = self.data.detector.Eth / self.Zdet
+            rigs = self.data.uhecr.rigidity if self.analysis_type == self.GMFdirection_type else self.data.uhecr.energy / self.Zdet
+            self.fit_input["Rdet"] = rigs
+            self.fit_input["Rth"] = self.data.detector.Eth / self.Zdet  # should modify this for directional model?
```

Comment:
Does this still need modifying as the comment suggested?

Reply:
I think it's not necessary, as initially I thought that the rigidity would vary as the UHECRs are backtracked which would modify the energy threshold. But we assume rigidity conservation so the same Rth at Earth will hold in the galactic boundary.